scheduleJob with replace = true causes data corruption when new trigger type mismatch original trigger

Using qrtz version 2.3.2, with postgres. 

If a job & trigger exists, and a call is made to `scheduler.scheduleJob(job, Set.of(newTrigger), true)` where `newTrigger` has a different type (e.g. cron vs simple) compared to the existing trigger.

From what I can tell, the db query becomes an update instead of deleting  from the previous table (qrtz_simple_triggers) and inserting into the new one (qrtz_cron_triggers), BUT the trigger_type of `qrtz_triggers` is still updated from `SIMPLE` to `CRON`. Any subsequent reads will cause an error because no corresponding value exists in the `CRON` table.

# Rough configs
```
org.quartz.scheduler.instanceName = test-quartz
org.quartz.jobStore.isClustered = true
org.quartz.scheduler.instanceId = AUTO
org.quartz.threadPool.threadCount = 1
org.quartz.jobStore.class = org.quartz.impl.jdbcjobstore.JobStoreTX
org.quartz.jobStore.driverDelegateClass = org.quartz.impl.jdbcjobstore.PostgreSQLDelegate
...+ your db settings...
```

# Sample test code
```
    @Test
    public void existingTriggerDifferentType() throws SchedulerException {
        JobDetail job = newJob(HelloJob.class)
            .withIdentity("job1", "group1")
            .build();

        Trigger trigger = newTrigger()
            .withIdentity("trigger1", "group1")
            .startAt(Date.from(Instant.now().plusSeconds(100000)))
            .withSchedule(simpleSchedule()
                .withIntervalInSeconds(400)
                .repeatForever())
            .build();

        scheduler.scheduleJob(job, trigger);

        Trigger trigger2 = newTrigger()
            .withIdentity(trigger.getKey())
            .startNow()
            .withSchedule(CronScheduleBuilder.dailyAtHourAndMinute(10, 15))
            .build();

        scheduler.scheduleJob(job, Set.of(trigger2), true);
        scheduler.getTrigger(trigger.getKey());
    }

    public static class HelloJob implements Job {
        @Override
        public void execute(final JobExecutionContext context) throws JobExecutionException {
        }
    }
```

# Exception
```
org.quartz.JobPersistenceException: Couldn't retrieve trigger: No record found for selection of Trigger with key: 'group1.trigger1' and statement: SELECT * FROM QRTZ_CRON_TRIGGERS WHERE SCHED_NAME = 'test-quartz__6e549c21-6dba-4442-ae4a-fb32a442da8e' AND TRIGGER_NAME = ? AND TRIGGER_GROUP = ?
 [See nested exception: java.lang.IllegalStateException: No record found for selection of Trigger with key: 'group1.trigger1' and statement: SELECT * FROM QRTZ_CRON_TRIGGERS WHERE SCHED_NAME = 'test-quartz__6e549c21-6dba-4442-ae4a-fb32a442da8e' AND TRIGGER_NAME = ? AND TRIGGER_GROUP = ?]
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.retrieveTrigger(JobStoreSupport.java:1538)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport$12.execute(JobStoreSupport.java:1527)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.executeInNonManagedTXLock(JobStoreSupport.java:3864)
	at org.quartz.impl.jdbcjobstore.JobStoreTX.executeInLock(JobStoreTX.java:93)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.executeWithoutLock(JobStoreSupport.java:3800)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.retrieveTrigger(JobStoreSupport.java:1524)
	at org.quartz.core.QuartzScheduler.getTrigger(QuartzScheduler.java:1505)
	at org.quartz.impl.StdScheduler.getTrigger(StdScheduler.java:508)
...
Caused by: java.lang.IllegalStateException: No record found for selection of Trigger with key: 'group1.trigger1' and statement: SELECT * FROM QRTZ_CRON_TRIGGERS WHERE SCHED_NAME = 'test-quartz__6e549c21-6dba-4442-ae4a-fb32a442da8e' AND TRIGGER_NAME = ? AND TRIGGER_GROUP = ?
	at org.quartz.impl.jdbcjobstore.CronTriggerPersistenceDelegate.loadExtendedTriggerProperties(CronTriggerPersistenceDelegate.java:107)
	at org.quartz.impl.jdbcjobstore.StdJDBCDelegate.selectTrigger(StdJDBCDelegate.java:1819)
	at org.quartz.impl.jdbcjobstore.JobStoreSupport.retrieveTrigger(JobStoreSupport.java:1536)
	... 40 more
```

# DB before update
```
pge_ywebyilapeco=# select * from qrtz_triggers;
                    sched_name                     | trigger_name | trigger_group | job_name | job_group | description | next_fire_time | prev_fire_time | priority | trigger_state | trigger_type |  start_time   | end_time | calendar_name | misfire_instr | job_data
---------------------------------------------------+--------------+---------------+----------+-----------+-------------+----------------+----------------+----------+---------------+--------------+---------------+----------+---------------+---------------+----------
 test-quartz__8636ff67-628b-4701-9e7e-64c59bc23a13 | trigger1     | group1        | job1     | group1    |             |  1685151581209 |             -1 |        5 | WAITING       | SIMPLE       | 1685151581209 |        0 |               |             0 | \x
(1 row)

pge_ywebyilapeco=# select * from qrtz_simple_triggers;
                    sched_name                     | trigger_name | trigger_group | repeat_count | repeat_interval | times_triggered
---------------------------------------------------+--------------+---------------+--------------+-----------------+-----------------
 test-quartz__8636ff67-628b-4701-9e7e-64c59bc23a13 | trigger1     | group1        |           -1 |          400000 |               0
(1 row)

pge_ywebyilapeco=# select * from qrtz_cron_triggers;
 sched_name | trigger_name | trigger_group | cron_expression | time_zone_id
------------+--------------+---------------+-----------------+--------------
(0 rows)
```

# DB after update
```
pge_ywebyilapeco=# select * from qrtz_triggers;
                    sched_name                     | trigger_name | trigger_group | job_name | job_group | description | next_fire_time | prev_fire_time | priority | trigger_state | trigger_type |  start_time   | end_time | calendar_name | misfire_instr | job_data
---------------------------------------------------+--------------+---------------+----------+-----------+-------------+----------------+----------------+----------+---------------+--------------+---------------+----------+---------------+---------------+----------
 test-quartz__8636ff67-628b-4701-9e7e-64c59bc23a13 | trigger1     | group1        | job1     | group1    |             |  1685096100000 |             -1 |        5 | WAITING       | CRON         | 1685051583000 |        0 |               |             0 | \x
(1 row)

pge_ywebyilapeco=# select * from qrtz_cron_triggers;
 sched_name | trigger_name | trigger_group | cron_expression | time_zone_id
------------+--------------+---------------+-----------------+--------------
(0 rows)

pge_ywebyilapeco=# select * from qrtz_simple_triggers;
                    sched_name                     | trigger_name | trigger_group | repeat_count | repeat_interval | times_triggered
---------------------------------------------------+--------------+---------------+--------------+-----------------+-----------------
 test-quartz__8636ff67-628b-4701-9e7e-64c59bc23a13 | trigger1     | group1        |           -1 |          400000 |               0
(1 row)
```

--- a/quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java
+++ b/quartz/src/main/java/org/quartz/impl/jdbcjobstore/StdJDBCDelegate.java
@@ -1232,11 +1232,17 @@ public class StdJDBCDelegate implements DriverDelegate, StdJDBCConstants {
             }
 
             insertResult = ps.executeUpdate();
-            
+
+            // Extended properties live in type-specific tables (SIMPLE/CRON/...).
+            // updateExtendedTriggerProperties only updates the new type's table, so a
+            // replace that changes trigger type (e.g. SIMPLE -> CRON) leaves the old
+            // extension row behind and inserts nothing into the new table. Always
+            // replace extended rows via delete + insert. See #920.
+            deleteTriggerExtension(conn, trigger.getKey());
             if(tDel == null)
-                updateBlobTrigger(conn, trigger);
+                insertBlobTrigger(conn, trigger);
             else
-                tDel.updateExtendedTriggerProperties(conn, trigger, state, jobDetail);
+                tDel.insertExtendedTriggerProperties(conn, trigger, state, jobDetail);
             
         } finally {
             closeStatement(ps);

--- a/quartz/src/test/java/org/quartz/AbstractJobStoreTest.java
+++ b/quartz/src/test/java/org/quartz/AbstractJobStoreTest.java
@@ -749,6 +749,55 @@ public abstract class AbstractJobStoreTest  {
     }
 
 
+    /**
+     * Replacing an existing trigger with one of a different concrete type must
+     * rewrite extended trigger rows (e.g. SIMPLE -> CRON). Otherwise the base
+     * QRTZ_TRIGGERS.TRIGGER_TYPE is updated while the old extension row remains
+     * and the new type's extension table is empty, corrupting subsequent reads.
+     *
+     * @see <a href="https://github.com/quartz-scheduler/quartz/issues/920">#920</a>
+     */
+    @Test
+    void testStoreTriggerReplaceChangesTriggerType() throws Exception {
+        Date start = new Date();
+        OperableTrigger simple = new SimpleTriggerImpl(
+                "typeChangeTrigger", "typeChangeGroup",
+                this.fJobDetail.getName(), this.fJobDetail.getGroup(),
+                start, null, SimpleTrigger.REPEAT_INDEFINITELY, 400_000L);
+        this.fJobStore.storeTrigger(simple, false);
+
+        OperableTrigger loadedSimple = this.fJobStore.retrieveTrigger(simple.getKey());
+        assertNotNull(loadedSimple);
+        assertTrue(loadedSimple instanceof SimpleTrigger);
+
+        OperableTrigger cron = (OperableTrigger) TriggerBuilder.newTrigger()
+                .withIdentity(simple.getKey())
+                .forJob(this.fJobDetail)
+                .withSchedule(CronScheduleBuilder.cronSchedule("0 0 12 * * ?"))
+                .startAt(start)
+                .build();
+
+        assertDoesNotThrow(() -> this.fJobStore.storeTrigger(cron, true));
+
+        OperableTrigger reloaded = this.fJobStore.retrieveTrigger(simple.getKey());
+        assertNotNull(reloaded, "trigger must remain readable after type-changing replace");
+        assertTrue(reloaded instanceof CronTrigger, "replaced trigger should be stored as CronTrigger");
+        assertEquals("0 0 12 * * ?", ((CronTrigger) reloaded).getCronExpression());
+
+        // Reverse direction: CRON -> SIMPLE
+        OperableTrigger simpleAgain = new SimpleTriggerImpl(
+                simple.getKey().getName(), simple.getKey().getGroup(),
+                this.fJobDetail.getName(), this.fJobDetail.getGroup(),
+                start, null, 3, 60_000L);
+        assertDoesNotThrow(() -> this.fJobStore.storeTrigger(simpleAgain, true));
+
+        OperableTrigger reloadedSimple = this.fJobStore.retrieveTrigger(simple.getKey());
+        assertNotNull(reloadedSimple);
+        assertTrue(reloadedSimple instanceof SimpleTrigger);
+        assertEquals(3, ((SimpleTrigger) reloadedSimple).getRepeatCount());
+        assertEquals(60_000L, ((SimpleTrigger) reloadedSimple).getRepeatInterval());
+    }
+
     @Test
     void testStoreJobReplacesJob() throws Exception {
 

